ibis/spark/ddl.py

```diff
@@ -0,0 +1,267 @@
+
+
+class CreateTable(impala_ddl.CreateTable):
+
+    """Create a table"""
+
+    def __init__(
+        self,
+        table_name,
+        database=None,
+        format='parquet',
+        can_exist=False,
+        path=None,
+        tbl_properties=None,
+    ):
+        super().__init__(
+            table_name,
+            database=database,
+            external=False,
+            format=format,
+            can_exist=can_exist,
+            partition=None,
+            path=path,
+            tbl_properties=tbl_properties,
+        )
+
+    def _storage(self):
+        # By the time we're here, we have a valid format
+        return 'USING {}'.format(self.format)
+
+
+class CreateTableWithSchema(CreateTable):
+    def __init__(self, table_name, schema, **kwargs):
+        super().__init__(table_name, **kwargs)
+        self.schema = schema
+
+    @property
+    def _pieces(self):
+        yield format_schema(self.schema)
+        yield self._storage()
+        yield self._location()
+
+
+class CTAS(CreateTable):
+
+    """
+    Create Table As Select
+    """
+
+    def __init__(
+        self,
+        table_name,
+        select,
+        database=None,
+        format='parquet',
+        can_exist=False,
+        path=None,
+    ):
+        super().__init__(
+            table_name,
+            database=database,
+            format=format,
+            can_exist=can_exist,
+            path=path,
+        )
+        self.select = select
+
+    @property
+    def _pieces(self):
+        yield self._storage()
+        yield self._location()
+        yield 'AS'
+        yield self.select.compile()
+
+
+class CreateView(CTAS):
+
+    """Create a view"""
+
+    def __init__(
+        self,
+        table_name,
+        select,
+        database=None,
+        or_replace=True,
+        temporary=False,
+    ):
+        super().__init__(
+            table_name,
+            select,
+            database=database,
+            can_exist=or_replace,
+        )
+        self.temporary = temporary
+
+    @property
+    def _pieces(self):
+        yield 'AS'
+        yield self.select.compile()
+
+    @property
+    def _prefix(self):
+        return 'CREATE {}{}VIEW'.format(
+            self._or_replace_clause(),
+            self._temporary_clause()
+        )
+
+    def _or_replace_clause(self):
+        return 'OR REPLACE ' if self.can_exist else ''
+
+    def _temporary_clause(self):
+        return 'TEMPORARY ' if self.temporary else ''
+
+    def _if_exists(self):
+        return ''
+
+
+def format_schema(schema):
+    elements = [
+        _format_schema_element(name, t)
+        for name, t in zip(schema.names, schema.types)
+    ]
+    return '({})'.format(',\n '.join(elements))
+
+
+def _format_schema_element(name, t):
+    return '{} {}'.format(
+        quote_identifier(name, force=True), _type_to_sql_string(t)
+    )
+
+
+class CreateDatabase(impala_ddl.CreateDatabase):
+    pass
+
+
+class DropObject(impala_ddl.DropObject):
```

Comment:
Is there a reason to define this? It looks like it's only used in `DropDatabase`. Why not just directly subclass `impala_ddl.DropObject` in the definition of `DropDatabase`?

Reply:
Also, it looks like `DropObject` is a parent of `DropTable` in `impala_ddl`. Did you mean to do that in this file as well?